### DISCUSSION TITLE
Can I adjust the granularity of the bucket in Istio metrics?

### DISCUSSION THREAD

**Discussion by xjtupdy** [istio/istio#49424] [Q&A]
Some of Istio service-level metrics, such as _istio_request_duration_milliseconds_ are _DISTRIBUTION_ . The division of the bucket is fine-grained. I want to know if I can adjust the granularity by EnvoyFilter or other approaches.

**Comment by zirain** [comment]
https://istio.io/latest/docs/reference/config/annotations/#SidecarStatsHistogramBuckets

**Reply by xjtupdy** [reply]
Thanks for your reply.
But I can't find it in doc until v1.19, so the annotation can be used since v1.19?
Can I make it when using previous version?

**Reply by zirain** [reply]
you can use custom bootstrap

**Reply by xjtupdy** [reply]
Maybe I can change bootstrap by using "proxyBootstrapTemplatePath". 
Where can I find the default proxy Bootstrap Template?

**Reply by zirain** [reply]
[here](https://github.com/istio/istio/tree/b6fc1fb732a5541624869141abb3ae3443044300/samples/custom-bootstrap) is a demo.

[this](https://github.com/istio/istio/blob/1e9dd6097865b15bc051c8c17e621cbfe56ad157/tools/packaging/common/envoy_bootstrap.json) is the template, be careful it's may not same across different versions.
